src/services/upload.ts

```diff
@@ -0,0 +1,125 @@
+import { storage } from "~/firebase/init";
+import {
+  ref,
+  uploadBytes,
+  getDownloadURL,
+  FirebaseStorage,
+  listAll,
+  StorageReference,
+  uploadBytesResumable,
+  deleteObject,
+} from "firebase/storage";
+import { Dispatch } from "react";
+
+type UploadFilePath = {
+  userId: string;
+  file: File;
+};
+type UpdateFile = {
+  filePath: string;
+  file: File;
+};
+
+class UploadService {
+  public readonly rootFolder: string = "jazzapp/music/";
```

Comment:
Your app may want to store anything except tracks, you'll want to use another folder. I think you should use a constructor here to provide a subfolder.

```suggestion
class UploadService {
  public static readonly rootFolder: string = "jazzapp/";

  public readonly namespace: string;

  constructor(namespace: string) {
    this.namespace = namespace;
  }

  get folder() {
    return `${UploadService.rootFolder}${this.namespace}/`;
  }
```

Then use the `folder` variable in this class, not `rootFolder`.

Also, now you'll be able to create:

```tsx
const TrackUploadService = new UploadService("music")
```